ford/md_extensions.py

```diff
@@ -0,0 +1,307 @@
+        self._find_admonitions()
+        self._process_admonitions()
+        return self.lines
+
+    def _find_admonitions(self):
+        """Scans the lines to search for admonitions."""
+        self.admonitions = []
+        current_admonition = None
+
+        for idx, line in enumerate(self.lines):
+            match = self.ADMONITION_RE.search(line)
+
+            if match and not match.group("end"):
+                if current_admonition:
+                    if not current_admonition.end_idx:
+                        current_admonition.end_idx = idx
+                    self.admonitions.append(current_admonition)
+                current_admonition = self.Admonition(
+                    type=match.group("type"),
+                    start_idx=idx,
+                )
+
+            elif match and match.group("end"):
+                if current_admonition:
+                    if match.group("type").lower() != current_admonition.type.lower():
+                        pass  # error: type of start and end marker dont' match
+                    current_admonition.end_idx = idx
+                    self.admonitions.append(current_admonition)
+                    current_admonition = None
+                else:
+                    pass  # error: end marker found without start marker
+
+            elif line == "" and current_admonition and not current_admonition.end_idx:
+                # empty line encountered while in an admonition. We set end_line but don't
+                # move it to the list yet since an end marker (@end...) may follow
+                # later.
+                current_admonition.end_idx = idx
+
+        if current_admonition:
+            # We reached the last line and the last admonition wasn't moved to the list yet.
+            if not current_admonition.end_idx:
+                current_admonition.end_idx = idx
+            self.admonitions.append(current_admonition)
+
+    def _process_admonitions(self):
+        """Processes the admonitions to convert the lines to the markdown syntax."""
+
+        # We handle the admonitions in the reverse order since
+        # we may be deleting lines.
+        for admonition in self.admonitions[::-1]:
+
+            # last line--deal with possible text before or after end marker
+            idx = admonition.end_idx
+            m = self.ADMONITION_RE.search(self.lines[idx])
+            if m:
+                if m.group("posttxt"):
+                    self.lines.insert(idx + 1, m.group("posttxt"))
+                del self.lines[idx]
+                if m.group("pretxt"):
+                    self.lines.insert(idx, " " * self.INDENT_SIZE + m.group("pretxt"))
+            if self.lines[idx] == "":
+                del self.lines[idx]
+
+            # intermediate lines
+            for idx in range(admonition.start_idx + 1, admonition.end_idx):
+                if self.lines[idx] != "":
+                    self.lines[idx] = " " * self.INDENT_SIZE + self.lines[idx]
```

Comment:
With the changes above, this could be
```suggestion
                self.lines[idx] = textwrap.indent(self.lines[idx], self.INDENT)
```